### PR TITLE
4 user registration page

### PR DESCRIPTION
**Pull request by topher-nullset**
## Describe your changes
Adds create logic for users
   adds private error message handler to concatenate errors

## Relevant Issues

Closes #(issue)

## Checklists before requesting a review
**Testing**
- [x] I have written thorough tests that prove my fix is effective or that my feature works
- [x] All tests are passing

**Reviews**
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code has no unused/commented out code
- [x] I have tagged all necessary teammates as Reviewers
- [x] I have linked this PR to any relevant project tasks
- [x] I have added all relevant labels to this PR

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -11,6 +11,11 @@
   </head>
 
   <body>
+    <% flash.each do |type, msg| %>
```
Comment (ethanrossblack) [inline app/views/layouts/application.html.erb]: Where is `type` referenced in this flash .each block? Is it necessary?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,12 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+      record.errors.add attribute, options[:message] || "'#{value}' is not a valid email"
+    end
+  end
+end
```
Comment (ethanrossblack) [inline app/models/user.rb:7]: This class looks cool, but idk if the top of the `models/user.rb` is the right place for this.

Maybe we ask Mike about where we should put this?

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+<section id="registration-form">
+  <%= form_with url: users_path, local: true do |f| %>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+    <%= f.label :email %>
+    <%= f.text_field :email %>
+    <%= f.submit "Create New User" %>
+  <% end %>
+</section>
```
Comment (ethanrossblack) [inline app/views/users/new.html.erb:11]: nice forming

File: spec/features/users/new_spec.rb
Comment (ethanrossblack) [inline spec/features/users/new_spec.rb:1]: thanks for editing all of those flash messages

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (topher-nullset) [reply app/views/layouts/application.html.erb]: nope, "type" was unnecessary and i accidentally copied it over from my other project.

**Review by ethanrossblack — APPROVED** [review]
Looks good!

**Pull request merged**